src/realm/sync/transform.cpp

```diff
@@ -997,16 +997,18 @@ struct MergeUtils {
     bool same_path_element(const Instruction::Path::Element& left,
                            const Instruction::Path::Element& right) const noexcept
     {
-        const auto& pred = util::overload{
-            [&](uint32_t lhs, uint32_t rhs) {
+        auto pred = util::overload{
```

Comment:
This might be worth a changelog entry, in case it did fix undefined behaviour on windows?